dashboard/src/components/CollapsingMenu/MenuItemBase.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PropTypes from "prop-types";
+import warning from "warning";
+
+class CollapsingMenuItemBase extends React.PureComponent {
+  static propTypes = {
+    renderMenuItem: PropTypes.node.isRequired,
+    renderButton: PropTypes.node.isRequired,
+    renderAs: PropTypes.oneOf(["button", "menu-item", null]),
+  };
+
+  static defaultProps = {
+    renderAs: null,
+  };
+
+  render() {
+    const { renderMenuItem, renderButton, renderAs } = this.props;
+    warning(
```

Comment:
I strongly advise against using `warning`. It is only small part of the system facebook uses to "minify" errors in react.